sources/web/datalab/polymer/modules/github-file-manager/github-file-manager.ts

```diff
@@ -0,0 +1,268 @@
+      },
+    };
+    return this._getGithubApiManager().sendRequestAsync(restUrl, options, false);
+  }
+
+  private _getGithubApiManager(): GithubApiManager {
+    if (!this._githubApiManager) {
+      this._githubApiManager = new GithubApiManager();
+    }
+    return this._githubApiManager;
+  }
+
+  private _ghRootDatalabFile(): DatalabFile {
+    const path = '/';
+    return new GithubFile({
+      icon: '',
+      id: new DatalabFileId(path, FileManagerType.GITHUB),
+      name: '/',
+      status: DatalabFileStatus.IDLE,
+      type: DatalabFileType.FILE,
+    } as DatalabFile);
+  }
+
+  private _ghReposResponseToDatalabFiles(response: GhRepoResponse[]):
+      DatalabFile[] {
+    return response.map(repo => this._ghRepoToDatalabFile(repo));
+  }
+
+  private _ghDirEntriesResponseToDatalabFiles(response: GhDirEntryResponse[]):
+      DatalabFile[] {
+    return response.map(file => this._ghDirEntryToDatalabFile(file));
+  }
+
+  private _ghRepoToDatalabFile(repo: GhRepoResponse): DatalabFile {
+    const type = DatalabFileType.DIRECTORY;
+    const icon = Utils.getItemIconString(type);
+    return new GithubFile({
+      icon,
+      id: new DatalabFileId(repo.full_name, FileManagerType.GITHUB),
+      name: repo.name,
+      status: DatalabFileStatus.IDLE,
+      type,
+    } as DatalabFile);
+  }
+
+  private _ghDirEntryToDatalabFile(file: GhDirEntryResponse): DatalabFile {
+    const type =
+        file.name.endsWith('.ipynb') ? DatalabFileType.NOTEBOOK :
+        file.type === 'dir' ? DatalabFileType.DIRECTORY :
+        DatalabFileType.FILE;
+    const icon = Utils.getItemIconString(type);
+    const pathParts = file.url.split('/');
+    const prefix = pathParts.slice(4, 6).join('/'); // user and project
+    const path = prefix + '/' + file.path;
+    return new GithubFile({
+      icon,
+      id: new DatalabFileId(path, FileManagerType.GITHUB),
+      name: file.name,
+      status: DatalabFileStatus.IDLE,
+      type,
+    } as DatalabFile);
+  }
+
+  private _ghFileToDatalabFile(file: GhFileResponse): DatalabFile {
+    const type = file.type === 'dir' ?
+        DatalabFileType.DIRECTORY : DatalabFileType.FILE;
+    const icon = Utils.getItemIconString(type);
+    const pathParts = file.url.split('/');
+    const prefix = pathParts.slice(4, 6).join('/'); // user and project
+    const path = prefix + '/' + file.path;
+    return new GithubFile({
+      icon,
+      id: new DatalabFileId(path, FileManagerType.GITHUB),
+      name: file.name,
+      status: DatalabFileStatus.IDLE,
+      type,
+    } as DatalabFile);
+  }
+
+  private _ghFileToContentString(file: GhFileResponse): string {
+    if (file.encoding != 'base64') {
+      throw new Error('github file encoding "' + file.encoding +
+        '" is not supported');
+    }
+    return atob(file.content);
+  }
+}
+
+// We just want the sendRequestAsync method of BaseApimanager.
+class GithubApiManager extends BaseApiManager {
```

Comment:
Do we need this if we specify `false` for `prependBasepath` when calling `sendRequestAsync`?